diff --git a/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000-ubootmod.dts b/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000-ubootmod.dts
index f26ab1a..7c75794 100644
--- a/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000-ubootmod.dts
+++ b/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000-ubootmod.dts
@@ -13,15 +13,33 @@
 	bootargs-append = " root=/dev/fit0 rootwait";
 };
 
+&spi_nand_flash {
+	mediatek,nmbm;
+	mediatek,bmt-max-ratio = <1>;
+	mediatek,bmt-max-reserved-blocks = <64>;
+};
+
 &partitions {
 	partition@580000 {
+		label = "crash";
+		reg = <0x580000 0x40000>;
+		read-only;
+	};
+
+	partition@5c0000 {
+		label = "crash_log";
+		reg = <0x5c0000 0x40000>;
+		read-only;
+	};
+
+	partition@600000 {
 		compatible = "linux,ubi";
-		reg = <0x580000 0x7a80000>;
+		reg = <0x600000 0x1da00000>;
 		label = "ubi";
 
 		volumes {
-			ubi_rootdisk: ubi-volume-fit {
-				volname = "fit";
+			ubi_rootdisk: ubi-volume-kernel {
+				volname = "kernel";
 			};
 		};
 	};
diff --git a/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000.dtsi b/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000.dtsi
index 0991034..1f037f1 100644
--- a/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000.dtsi
+++ b/target/linux/mediatek/dts/mt7986a-xiaomi-redmi-router-ax6000.dtsi
@@ -22,7 +22,7 @@
 	};
 
 	memory@40000000 {
-		reg = <0 0x40000000 0 0x20000000>;
+		reg = <0 0x40000000 0 0x80000000>;
 		device_type = "memory";
 	};
 
